utils/near-cache/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+use cached::{Cached, SizedCache};
+use std::hash::Hash;
+use std::sync::Mutex;
+
+#[allow(unused)]
+struct MyCache<K, V> {
+    inner: Mutex<SizedCache<K, V>>,
+}
+
+// MyCache<CryptoHash, Result<wasmer_runtime::Module, CompilationError>>
+impl<K, V> MyCache<K, V>
+where
+    K: Hash + Eq + PartialEq + Clone,
+    V: Clone,
+{
+    #[allow(unused)]
+    pub fn new(capacity: usize) -> Self {
+        Self { inner: Mutex::new(SizedCache::<K, V>::with_size(capacity)) }
+    }
+
+    #[allow(unused)]
+    pub fn get_from_cache_or_insert<F>(&self, key: K, f: F) -> V
+    where
+        F: FnOnce(&K) -> V,
+        V: Clone,
+    {
+        if let Some(result) = self.get_from_cache(&key) {
+            return result.clone();
+        }
+        let val = f(&key);
+        self.inner.lock().unwrap().cache_set(key, val.clone());
```

Comment:
Micro-optimizatin: let's clone the value before locking the lock, to avoid allocations under lock: 

```
let val = f(&key);
let val_clone = val.clone();
self.inner.lock().unwrap().cache_set(key, val_clone);
val
```